commands/item.js

```diff
@@ -38,24 +40,60 @@ Examples:
 
   let temp = argv._;
   temp = temp.filter(x => x !== 'item');
-  const itemName = temp.join(' ');
+  let itemName = temp.join(' ');
+
+  let candidates = resolveItemName(itemName); // check item name against local static array 
+  if (Object.prototype.toString.call(candidates) === '[object Array]') { // only continue if resolveItemName returned an array
+    if (candidates.length == 1) { // unambiguous match, replace input
+      itemName = candidates[0]; 
+    }
+    else if (candidates.length > 1) { // multiple matches, suggest some candidates
+      const maxCandidates = 5;
+      let suggestions = candidates.slice(0, maxCandidates);
+      if (candidates.length > maxCandidates) 
+        suggestions.push('...');
+      suggestions = suggestions.join("\n");
+      let embed = new Discord.RichEmbed()
+        .setTitle(`Multiple results for '${itemName}'`)
+        .addField('Did you mean:', `${suggestions}`)
+        .attachFile(sherlockAttachment)
+        .setThumbnail('attachment://sherlock.png');
+      return { embed }; 
+    }
+  }
 
   let param = '';
   if (argv.a) {
-    param = `?quality=ARTIFACT${ argv.a }`;
+    param = `?quality=ARTIFACT${argv.a}`;
   }
 
   let url = 'https://questland-public-api.cfapps.io/items/name/'
     + encodeURIComponent(itemName)
     + param;
   const response = await fetch(url);
-  return response.ok ? printItem(await response.json()) : 'Unable to locate item.';
+  return response.ok ? printItem(await response.json()) : new Discord.RichEmbed()
+    .setTitle(`Item '${itemName}' not found`)
+    .addField('Please check your input', '\u200b')
+    .attachFile(noideaAttachment)
+    .setThumbnail('attachment://noidea.png');
 });
 
+/// match an item name against the static array of available item names
+const resolveItemName = (name) => {
+  try {
+    let module = require('../data/itemNames.js');
+    let itemNames = module.itemNames;
+    return itemNames.filter(i => i.toLowerCase().includes(name.toLowerCase()));
```

Comment:
I think it would be pretty low lift to apply a sort here that anything that "starts with" the partial name would rise to the top of the suggestions. Might help make the suggestions slightly better

Reply:
Good point. Was on my mind, too, but then forgot about it again

Reply:
Implemented